.github/workflows/macos_tests.yaml

```diff
@@ -35,6 +37,15 @@ jobs:
         with:
           submodules: "recursive"
 
+      - name: Checkout macOS SDK
+        uses: actions/checkout@v4
+        with:
+          repository: alexey-lysiuk/macos-sdk
+          path: 3rdparty/macos-sdk
+          ref: main
+          sparse-checkout: MacOSX15.5.sdk/
+      - run: echo "SDKROOT=$(find ${{ github.workspace }}/3rdparty/macos-sdk -name 'SDKSettings.plist' -printf '%h\n')" >> $GITHUB_ENV
+
```

Comment:
Just a nit: Should we maybe use taskcluster artifact here?

Reply:
I preferred to do it this way as it kinda keeps things in the Github ecosystem.

My first attempt at this was to fetch the SDK directly from Apple, but it seems that they block the URL with a 403 unless it comes from an authorized location (Taskcluster is authorized but GHA is not).

Reply:
Thats fair then :)